### DISCUSSION TITLE
Typing difference of `params` parameter between `Session.scalars` and `AsyncSession.scalars`

### DISCUSSION THREAD

**Discussion by provinzkraut** [sqlalchemy/sqlalchemy#9925] [Usage Questions] [answered]
In `Session.scalars`, `params` is typed as `Optional[_CoreAnyExecuteParams]`, but in `AsyncSession.scalars` it's `Optional[_CoreSingleExecuteParams]`. The underlying `.execute` however accepts `_CoreAnyExecuteParams` in both cases, and supplying a list of mappings seems to work just as well to `AsyncSession.scalars`. 

https://github.com/sqlalchemy/sqlalchemy/blob/5927b2d39acbb0cb71f396a28ddc76ca44322103/lib/sqlalchemy/orm/session.py#L2308-L2311

https://github.com/sqlalchemy/sqlalchemy/blob/5927b2d39acbb0cb71f396a28ddc76ca44322103/lib/sqlalchemy/ext/asyncio/session.py#L528-L531

Is this simply an error in the typing or is there a reason for this difference?

**Comment by zzzeek** [comment, marked as answer]
likely an oversight.   if you want to look over the synonymous methods in session/async and see if there are any other anomalies we can work up a PR to repair them at once.

**Reply by zzzeek** [reply]
the _CoreAnyExecuteParams is the correct one, that was changed fairly recently to serve the use case of an INSERT..RETURNING of ORM objects using bulk ORM insert.

**Reply by provinzkraut** [reply]
>  if you want to look over the synonymous methods in session/async and see if there are any other anomalies we can work up a PR to repair them at once.

Will do!
